### PR TITLE
Add Pearson hashes

### PR DESCRIPTION
**Pull request by data-man**
Acknowledgements to @Logan007.

(Slightly modified sources.)

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -719,6 +719,15 @@ HashInfo g_hashes[] =
   //{ wyhash_condom_test,   64, 0x7C62138D, "wyhash_condom",  "wyhash v3 condom 2 (64-bit)", GOOD, { } },
 #endif
   { nmhash32_test,        32, 0x0DC5C2DC, "nmhash32",         "nmhash32", GOOD, {}},
+#ifdef HAVE_INT64
+  { pearson64_test,       64, 0x12E4C8CD, "pearsonhash64",    "Pearson hash, 64-bit", GOOD, {}},
```
Comment (data-man) [inline main.cpp:723]: Is it GOOD? I'm not sure.

File: pearson_hash/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# Pearson Block Hashing
+
+The relatively low speed seen with regular [Pearson Hashing](https://github.com/Logan007/pearson) on modern CPUs might originate from the fact that it only processes one byte of input at a time. So, this is an attempt to read and digest multiple bytes, a block of bytes, in order to speed things up. Still relying on the underlying Pearson Hasing scheme, I tenderly call it **Pearson B. Hashing**.
```
Comment (rurban) [inline pearson_hash/README.md:3]: Typo: Hasing

File: pearson_hash/pearson.c [hunk 3]
```diff
@@ -0,0 +1,619 @@
+#include <stddef.h>
+#include <stdint.h>
+
+#include "pearson.h"
+
+// compile with 'LOW_MEM_FOOTPRINT' defined to make use of 256 byte look-up tabe only
```
Comment (rurban) [inline pearson_hash/pearson.c:6]: Typo: tabe

**Review by rurban — APPROVED** [review]
You might want to fix the minor typos

**Comment by rurban** [comment]
_mm_cvtsi128_si64 is not available with -m32. See travis

File: pearson_hash/README.md (hunk 2 above)
Comment (data-man) [reply pearson_hash/README.md]: Thank you!
I'll send a fixes to the original repository.

**Comment by rurban** [comment]
Basic Avalanche tests are failing. Not a good one

**Comment by rurban** [comment]
Done and fixed.

**Comment by data-man** [comment]
Thanks!

**Comment by Logan007** [comment]
@rurban
@data-man

I feel humbled to see the Pearson (B) Hash added to the venerable SMHasher. Thank you very much!

I found that **Pearson B**'s quality problems (Perlin, Seed) arise from not offering the opportunity to be seeded. I did some trials on a very natural way to add seeding and it seems to pass all tests. If I may, I would like to add a corresponding pull request.

Overall, I find it quite spurring to have the code added here and it might motivate me to think about some faster "Pearson C" :wink:

**Comment by rurban** [comment]
Logan oos Even ***@***.***> schrieb am Fr., 7. Mai 2021,
11:47:

> @rurban <https://github.com/rurban>
> @data-man <https://github.com/data-man>
>
> I feel humbled to see the Pearson (B) Hash added to the venerable
> SMHasher. Thank you very much!
>
> I found that *Pearson B*'s quality problems (Perlin, Seed) arise from not
> offering the opportunity to be seeded. I did some trials on a very natural
> way to add seeding and it seems to pass all tests. If I may, I would like
> to add a corresponding pull request.
>

Sure

Overall, I find it quite spurring to have the code added here and it might

**Pull request merged**